platforms/core-runtime/collections/src/main/java/org/gradle/internal/collect/ArrayCopy.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.collect;
+
+import java.util.Arrays;
+
+/// Different array copying operations required by the persistent collection
+/// implementations.
+///
+/// In general, these should be as efficient as possible, contain
+/// no branching code, and minimize [System] calls.
```

Comment:
What's wrong with System calls specifically compared to other code?

I see a lot of `System` usages in this class